### ISSUE TITLE
possible Bug with get/setStateAsync

### ISSUE THREAD

**Issue by Homoran** [ioBroker/ioBroker.javascript#777]
*the short program

```
on({id: "sonos.0.root.192_168_138_59.state", change: "ne"}, async function (obj) {
    console.log("getriggert");
    var zustand = await getStateAsync("sonos.0.root.192_168_138_59.state");
    console.log(JSON.stringify(zustand));
    if (await getStateAsync("sonos.0.root.192_168_138_59.state").val == "play") {
    console.log("Bedingung erkannt");
    await setStateAsync("sonos.0.root.192_168_138_59.state", "stop", false);
    console.log("ausgeführt");
    } else {
     console.log("Bedingung falsch");
     await setStateAsync("sonos.0.root.192_168_138_59.state", "pause", false);
     console.log("ausgeführt");
    }
});
```
does trigger, but always goes to condition -> false

js v5.0.2

**iobroker-log:**

```
2021-02-28 15:38:08.746 - info: sonos.0 (18554) try to control id sonos.0.root.192_168_138_59.play with {"val":true,"ack":false,"ts":1614523088736,"q":0,"from":"system.adapter.admin.0","user":"system.user.admin","lc":1614441319432}
--
2021-02-28 15:38:11.314 - info: javascript.0 (1547) script.js.Skript_1: getriggert
2021-02-28 15:38:11.500 - info: javascript.0 (1547) script.js.Skript_1: {"val":"play","ack":true,"ts":1614523091277,"q":0,"from":"system.adapter.sonos.0","user":"system.user.admin","lc":1614523091277}
2021-02-28 15:38:11.504 - info: javascript.0 (1547) script.js.Skript_1: Bedingung falsch
2021-02-28 15:38:11.539 - info: sonos.0 (18554) try to control id sonos.0.root.192_168_138_59.state with {"val":"pause","ack":false,"ts":1614523091530,"q":0,"c":"script.js.Skript_1","from":"system.adapter.javascript.0","user":"system.user.admin","lc":1614523091530}
2021-02-28 15:38:11.571 - info: javascript.0 (1547) script.js.Skript_1: getriggert
2021-02-28 15:38:11.580 - info: javascript.0 (1547) script.js.Skript_1: ausgeführt
2021-02-28 15:38:11.596 - info: javascript.0 (1547) script.js.Skript_1: {"val":"pause","ack":false,"ts":1614523091530,"q":0,"c":"script.js.Skript_1","from":"system.adapter.javascript.0","user":"system.user.admin","lc":1614523091530}
2021-02-28 15:38:11.603 - info: javascript.0 (1547) script.js.Skript_1: Bedingung falsch
2021-02-28 15:38:11.652 - info: sonos.0 (18554) try to control id sonos.0.root.192_168_138_59.state with {"val":"pause","ack":false,"ts":1614523091640,"q":0,"c":"script.js.Skript_1","from":"system.adapter.javascript.0","user":"system.user.admin","lc":1614523091530}
2021-02-28 15:38:11.666 - info: javascript.0 (1547) script.js.Skript_1: ausgeführt
2021-02-28 15:38:12.325 - info: javascript.0 (1547) script.js.Skript_1: getriggert
2021-02-28 15:38:12.478 - info: javascript.0 (1547) script.js.Skript_1: {"val":"stop","ack":true,"ts":1614523092182,"q":0,"from":"system.adapter.sonos.0","user":"system.user.admin","lc":1614523092182}
2021-02-28 15:38:12.482 - info: javascript.0 (1547) script.js.Skript_1: Bedingung falsch
2021-02-28 15:38:12.520 - info: javascript.0 (1547) script.js.Skript_1: getriggert
2021-02-28 15:38:12.522 - info: sonos.0 (18554) try to control id sonos.0.root.192_168_138_59.state with {"val":"pause","ack":false,"ts":1614523092507,"q":0,"c":"script.js.Skript_1","from":"system.adapter.javascript.0","user":"system.user.admin","lc":1614523092507}
2021-02-28 15:38:12.533 - info: javascript.0 (1547) script.js.Skript_1: {"val":"pause","ack":false,"ts":1614523092507,"q":0,"c":"script.js.Skript_1","from":"system.adapter.javascript.0","user":"system.user.admin","lc":1614523092507}
2021-02-28 15:38:12.542 - info: javascript.0 (1547) script.js.Skript_1: Bedingung falsch
2021-02-28 15:38:12.564 - info: javascript.0 (1547) script.js.Skript_1: ausgeführt
2021-02-28 15:38:12.644 - info: sonos.0 (18554) try to control id sonos.0.root.192_168_138_59.state with {"val":"pause","ack":false,"ts":1614523092636,"q":0,"c":"script.js.Skript_1","from":"system.adapter.javascript.0","user":"system.user.admin","lc":1614523092507}
2021-02-28 15:38:12.661 - info: javascript.0 (1547) script.js.Skript_1: ausgeführt
```

**Comment by Homoran** [comment]
script and resulting log exchanged

**Comment by GermanBluefox** [comment]
Fixed with 5.0.3
